packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2514,7 +2514,13 @@ export class SnapController extends BaseController<
       const origins = getRpcCaveatOrigins(rpcPermission);
       assert(origins);
 
-      if ((isSnap && !origins.snaps) || (!isSnap && !origins.dapps)) {
+      if (
+        !isOriginAllowed(
+          origins,
+          subject?.subjectType ?? SubjectType.Website,
```

Comment:
I'm curious if the subject can even be undefined. Maybe it is worth failing if it is? 🤔

Reply:
According to the type it can be. I don't know if we should fail if it is?

Reply:
Hmm, could be worth asserting, but we haven't so far, so maybe this is fine